src/jailer/src/env.rs

```diff
@@ -275,12 +287,56 @@ impl Env {
         // namespace. Safe because we are passing valid parameters.
         SyscallReturnCode(unsafe { libc::close(netns_fd) })
             .into_empty_result()
-            .map_err(Error::CloseNetNsFd)
+            .map_err(Error::CloseNetNsFd)?;
+
+        // Since namespaces are shared by default when creating a new process using fork or clone,
+        // unshare() is used to disassociate the current process from the default shared namespace.
+        SyscallReturnCode(unsafe { libc::unshare(libc::CLONE_NEWNS) })
```

Comment:
I would explicitly add in the comment that this unshares the _mount_ namespace.

Reply:
Done.